src/modules/commander/Commander.cpp

```diff
@@ -1832,78 +1831,80 @@ Commander::run()
-							if (TRANSITION_CHANGED == main_state_transition(status, commander_state_s::MAIN_STATE_AUTO_LOITER, status_flags,
-									&internal_state)) {
-								geofence_loiter_on = true;
-							}
+						break;
+					}
 
-							break;
+				case (geofence_result_s::GF_ACTION_RTL) : {
+						if (TRANSITION_CHANGED == main_state_transition(status, commander_state_s::MAIN_STATE_AUTO_RTL, status_flags,
+								&internal_state)) {
+							_geofence_rtl_on = true;
 						}
 
-					case (geofence_result_s::GF_ACTION_RTL) : {
-							if (TRANSITION_CHANGED == main_state_transition(status, commander_state_s::MAIN_STATE_AUTO_RTL, status_flags,
-									&internal_state)) {
-								geofence_rtl_on = true;
-							}
-
-							break;
-						}
+						break;
+					}
 
-					case (geofence_result_s::GF_ACTION_TERMINATE) : {
-							warnx("Flight termination because of geofence");
-							mavlink_log_critical(&mavlink_log_pub, "Geofence violation! Flight terminated");
-							armed.force_failsafe = true;
-							status_changed = true;
-							break;
-						}
+				case (geofence_result_s::GF_ACTION_TERMINATE) : {
+						warnx("Flight termination because of geofence");
+						mavlink_log_critical(&mavlink_log_pub, "Geofence violation! Flight terminated");
+						armed.force_failsafe = true;
+						status_changed = true;
+						break;
 					}
 				}
 			}
 
+			_geofence_violated_prev = geofence_result.geofence_violated;
+
 			// reset if no longer in LOITER or if manually switched to LOITER
-			geofence_loiter_on = geofence_loiter_on
-					     && (internal_state.main_state == commander_state_s::MAIN_STATE_AUTO_LOITER)
-					     && (sp_man.loiter_switch == manual_control_setpoint_s::SWITCH_POS_OFF
-						 || sp_man.loiter_switch == manual_control_setpoint_s::SWITCH_POS_NONE);
+			_geofence_loiter_on = _geofence_loiter_on
+					      && (internal_state.main_state == commander_state_s::MAIN_STATE_AUTO_LOITER)
+					      && (sp_man.loiter_switch == manual_control_setpoint_s::SWITCH_POS_OFF
+						  || sp_man.loiter_switch == manual_control_setpoint_s::SWITCH_POS_NONE);
 
 			// reset if no longer in RTL or if manually switched to RTL
-			geofence_rtl_on = geofence_rtl_on
-					  && (internal_state.main_state == commander_state_s::MAIN_STATE_AUTO_RTL)
-					  && (sp_man.return_switch == manual_control_setpoint_s::SWITCH_POS_OFF
-					      || sp_man.return_switch == manual_control_setpoint_s::SWITCH_POS_NONE);
+			_geofence_rtl_on = _geofence_rtl_on
+					   && (internal_state.main_state == commander_state_s::MAIN_STATE_AUTO_RTL)
+					   && (sp_man.return_switch == manual_control_setpoint_s::SWITCH_POS_OFF
+					       || sp_man.return_switch == manual_control_setpoint_s::SWITCH_POS_NONE);
+
+			_warning_action_on = _warning_action_on || (_geofence_loiter_on || _geofence_rtl_on);
 
-			warning_action_on = warning_action_on || (geofence_loiter_on || geofence_rtl_on);
+		} else {
+			// No geofence checks, reset flags
```

Comment:
This comment is wrong, right? There are still checks, there is just no violation.

Reply:
Correct :D

Reply:
Nope, this else is for "if not armed or no geofence action or in battery failsafe"

Reply:
I would understand this comment as "No geofence checks have been done, we reset the flags". Is that correct?

Reply:
Yes, correct